### ISSUE TITLE
Remove Modified FBuild Requirement

### ISSUE THREAD

**Issue by ClxS** [ClxS/FASTBuild-UE4#3]
A part of fixing the issue mentioned in https://github.com/ClxS/fastbuild-ue4/issues/1 will involve removing the misused DLL nodes. This would hopefully remove the required modification to FBuild which allows for EXE dependency nodes.

**Comment by liamkf** [comment]
Hi there! I was following the conversation about UE4 FASTBuild with interest, as we (@yass007) had made a version of our own last year to build Unreal with FASTBuild which we brought a bit more up to date for some new projects here: https://github.com/liamkf/Unreal_FASTBuild

It seems like we ran into some similar problems! When we were using it for Durango we had a few issues getting PCH and resources output names working correctly, as well as fun with Unreal's fondness for response files.

Along the way of hacking to get it working, we generally found it easier to modify FASTBuild rather than try to handle oddities in FASTBuild.cs. The version we posted doesn't require modifying FBuild or Unreal (other than calling it from ActionGraph.cs), but it's a bit more rickety that it would be otherwise.

Anyways... what may be of use for you! We used the Executable node for everything involving link.exe, and Library for lib.exe. It's lightly suspect in that it relies on /DLL being detected by FASTBuild in order to generate the DLLs, but it seems to work, so it may be useful to glance at. 

We did have to put in a 'dummy node' to make sure that dependencies buried in response files were waited for appropriately and had the right object names, which a "PrebuildDependencies" on the exe node would simplify. It looks like you guys avoided these by splitting the compile/link phases, which in retrospect... probably would have been a lot easier! :)

**Comment by ClxS** [comment]
Hi @liamkf 
I remember us having a few issues getting Durango working too. I had to use a fairly nasty looking hack to get it to be able to cache any of the Durango objects due to the /ZW argument. That hack's been included in FB 0.9 now though! 

Yeah splitting the Compile/Link phases was done out of necessity for us rather than design. I have added support for PreBuildDependencies to all of the node types in ours and did have a single-pass version working, but I'd ran into another issue which prevented us from using it on the project.

Thank you for sharing your approach! It'll help a lot. :)
